File: MDWSvistalayer/MDWS Source/mdo/mdo/src/mdo/domain/sm/trash/Surrogate.java
package gov.va.med.mhv.sm.model;

import gov.va.med.mhv.sm.enumeration.ParticipantTypeEnum;

import java.io.Serializable;
import java.util.Date;

public class Surrogate extends BaseModel implements Serializable
{

	private static final long serialVersionUID = 6814862738195064844L;
	private Long surrogateId;
	private User smsUser;
	private ParticipantTypeEnum surrogateType;
	private Date surrogateStartDate;
	private Date surrogateEndDate;
	private boolean surrogateAllDay;
	private Long timeZone;

	public User getSmsUser() {
		return smsUser;
	}

	public void setSmsUser(User smsUser) {
		this.smsUser = smsUser;
	}

	public boolean isSurrogateAllDay() {
		return surrogateAllDay;
	}

	public void setSurrogateAllDay(boolean surrogateAllDay) {
		this.surrogateAllDay = surrogateAllDay;
	}

	public Date getSurrogateEndDate() {
		return surrogateEndDate;
	}

	public void setSurrogateEndDate(Date surrogateEndDate) {
		this.surrogateEndDate = surrogateEndDate;
	}

	public Long getSurrogateId() {
		return surrogateId;
	}

	public void setSurrogateId(Long surrogateId) {
		this.surrogateId = surrogateId;
	}

	public Date getSurrogateStartDate() {
		return surrogateStartDate;
	}

	public void setSurrogateStartDate(Date surrogateStartDate) {
		this.surrogateStartDate = surrogateStartDate;
	}
	
	public boolean equals(Object a){
		try{
			Surrogate x = (Surrogate)a;
			return this.getId().equals(x.getId()); 
		}catch(Exception e){
			return false;
		}
	}
	
	public int hashCode(){
		return this.getId().hashCode();
	}

	public Long getTimeZone() {
		return timeZone;
	}

	public void setTimeZone(Long timeZone) {
		this.timeZone = timeZone;
	}

	public void setSurrogateType(ParticipantTypeEnum surrogateType) {
		this.surrogateType = surrogateType;
	}

	public ParticipantTypeEnum getSurrogateType() {
		return surrogateType;
	}
		
}

File: OtherComponents/MDWSvistalayer/MDWS Source/mdo/mdo/src/mdo/domain/sm/Surrogate.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace gov.va.medora.mdo.domain.sm
{
    [Serializable]
    public class Surrogate : BaseModel
    {
        private Int64 _surrogateId;

        public Int64 SurrogateId
        {
            get { return _surrogateId; }
            set { _surrogateId = value; }
        }
        private User _smsUser;

        public User SmsUser
        {
            get { return _smsUser; }
            set { _smsUser = value; }
        }
        //private ParticipantTypeEnum _surrogateType;

        //public ParticipantTypeEnum SurrogateType
        //{
        //    get { return _surrogateType; }
        //    set { _surrogateType = value; }
        //}
        private DateTime _surrogateStartDate;

        public DateTime SurrogateStartDate
        {
            get { return _surrogateStartDate; }
            set { _surrogateStartDate = value; }
        }
        private DateTime _surrogateEndDate;

        public DateTime SurrogateEndDate
        {
            get { return _surrogateEndDate; }
            set { _surrogateEndDate = value; }
        }
        private bool _surrogateAllDay;

        public bool SurrogateAllDay
        {
            get { return _surrogateAllDay; }
            set { _surrogateAllDay = value; }
        }
        private Int64 _timeZone;

        public Int64 TimeZone
        {
            get { return _timeZone; }
            set { _timeZone = value; }
        }
    }
}
